src/test/java/htsjdk/samtools/HtsgetBAMFileReaderTest.java

```diff
@@ -0,0 +1,255 @@
+package htsjdk.samtools;
+
+import htsjdk.HtsjdkTest;
+import htsjdk.samtools.util.CloseableIterator;
+import htsjdk.samtools.util.CoordMath;
+import org.testng.Assert;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+
+public class HtsgetBAMFileReaderTest extends HtsjdkTest {
```

Comment:
It would be good to run these tests with a variety of settings on the htsget reader.  Particularly I'd like to see them with async on for the htsget reader since it seems plausible that there could be potential issues there.  You could add a datprovider that has multiple readers.

The readers also need to be closed at the end.  

Typically I'm in favor of not reusing the same objects for multiple tests and generating them fresh each time, it helps avoid test ordering problems or if 1 test puts the objects into a broken state.  This should be OK though.

Reply:
The try with resources block will close the iterators. I rewrote them this way because I was actually having the exact problem you mentioned, where earlier tests failing before they closed the iterator caused later tests to fail. The HtsgetBAMFileReader doesn't have the "one iterator open at a time" restriction like the regular reader so that shouldn't happen. 

I'll add the provider with async io since I would've needed to anyway to test the prefetching